### ISSUE TITLE
[Bug] -nmap-cli fails when used combined with custom ports

### ISSUE THREAD

**Issue by dogasantos** [projectdiscovery/naabu#126]
**Describe the bug**
When using -nmap-cli option, the -p switch only works for full tcp "-p -". If you specify any port, it will fail.

naabu will throw an error related with interface:

`
[INF] Running nmap command: nmap -sV -oX naabu-output.xml -p  
Starting Nmap 7.91 ( https://nmap.org ) at 2020-12-29 10:58 EST
[ERR] Could not get network interfaces: exit status 1
`
However, provide the interface via `-interface` has no effect. 
Also, using -p through the nmap-cli won't work.

**To Reproduce**
naabu -iL ips.txt -nmap -p 80 -scan-all-ips -stats -nmap-cli 'nmap -sV -oX naabu-output.xml'

**Screenshots**
![naabu-working](https://user-images.githubusercontent.com/4156021/103297144-746c0f80-49d6-11eb-8e24-e2fed7d16b3b.PNG)

![naabu-not-working-1](https://user-images.githubusercontent.com/4156021/103297141-733ae280-49d6-11eb-87a4-0606a4127d4e.PNG)

![naabu-not-working-2](https://user-images.githubusercontent.com/4156021/103297143-746c0f80-49d6-11eb-8599-60e49a84bbff.PNG)

**Comment by bauthard** [comment]
@dogasantos Can you retry without `-nmap` and `-scan-all-ips` as those are unwanted for this can, `-scan-all-ips` is used when you scanning subs having multiple A record, and `-nmap` when you running **nmap** cmd using config file.

```bash
naabu -iL ip.txt -p 443,80 -stats -nmap-cli 'nmap -sV -oX naabu-output.xml'
                  __
  ___  ___  ___ _/ /  __ __
 / _ \/ _ \/ _ \/ _ \/ // /
/_//_/\_,_/\_,_/_.__/\_,_/ v2.0.3

		projectdiscovery.io

[WRN] Use with caution. You are responsible for your actions
[WRN] Developers assume no liability and are not responsible for any misuse or damage.
[INF] Running SYN scan with root privileges
[INF] Found 2 ports on host 104.16.100.52 (104.16.100.52)
104.16.100.52:443
104.16.100.52:80
[INF] Found 2 ports on host 104.16.99.52 (104.16.99.52)
104.16.99.52:80
104.16.99.52:443
[INF] Running nmap command: nmap -sV -oX naabu-output.xml -p 80,443 104.16.99.52,104.16.100.52
Starting Nmap 7.91 ( https://nmap.org ) at 2020-12-29 23:19 IST
[0:00:05] | Hosts: 3 | Ports: 2 | Retries: 3 | PPS: 4 | Packets: 24/18 (133%)
[0:00:10] | Hosts: 3 | Ports: 2 | Retries: 3 | PPS: 2 | Packets: 24/18 (133%)
[0:00:15] | Hosts: 3 | Ports: 2 | Retries: 3 | PPS: 1 | Packets: 24/18 (133%)
```

**Comment by dogasantos** [comment]
Hi,
still not working (same result)
I'm using same naabui version 2.0.3 and same nmap version 7.91.
That's weird

**Comment by bauthard** [comment]
Thanks, we will investigate this further.

**Comment by Shivangx01b** [comment]
I face same issue 

[INF] Running nmap command: nmap -sVC -Pn -oX /tmp/output.xml -p 80,443,8443,8080 104.16.100.52
Starting Nmap 7.91 ( https://nmap.org ) at 2021-01-04 11:46 UTC
[ERR] Could not get network interfaces: exit status 1

**Comment by dogasantos** [comment]
same happens when using naabu as a library.
For some reason, naabu is unable to find any open port (it doesn't run at all), then it has no ports to offer nmap as argument. 
Sample code:

`func runTcpScan(targetip string) {
	var options runner.Options
	//options.Silent = true
	options.Debug = true
	options.Ping = false
	options.ExcludeCDN = true
	options.Rate = 200
	options.Timeout = 6
	options.Retries = 3
	options.WarmUpTime = 5
	options.Host = targetip
	options.Interface = "enp1s0"
	options.InterfacesList = false
	options.TopPorts = "100"
	options.Threads = 6
	options.Nmap = true
	//options.Output = "/tmp/naabu-output.txt"
	options.NmapCLI = "nmap -sV -oX /tmp/nmap-output.xml --script=http-title,http-server-header,http-open-proxy,http-methods,http-headers,ssl-cert"

	naabuRunner, err := runner.NewRunner(&options)
	if err != nil {
		gologger.Fatal().Msgf("Could not create runner: %s\n", err)
	}
	err = naabuRunner.RunEnumeration()
	if err != nil {
		gologger.Fatal().Msgf("Could not run enumeration: %s\n", err)
	}

}`

======== 
result:

`
2021/07/20 21:52:15 New ipaddr to work with: 127.0.0.1
[INF] Running nmap command: nmap -sV -oX /tmp/nmap-output.xml --script=http-title,http-server-header,http-open-proxy,http-methods,http-headers,ssl-cert -p  
Starting Nmap 7.80 ( https://nmap.org ) at 2021-07-20 21:52 UTC
[ERR] Could not get network interfaces: exit status 1
`

Naabu own scan is being skipped, sometimes it runs but finds just one or 2 ports when it should find 15 open ports.
STRANGE THINGS

**Comment by dogasantos** [comment]
hate github code snippet.

**Comment by iwasd** [comment]
hi
I face same issue
naabu use v2.0.4

naabu -host 192.168.0.101 -p 443,80 -stats -nmap-cli 'nmap -sV -oX naabu-output.xml'

[INF] Running nmap command: nmap -sV -oX naabu-output.xml -p 443,80 192.168.0.101
Starting Nmap 7.91 ( https://nmap.org ) at 2021-07-29 12:19 UTC
[ERR] Could not get network interfaces: exit status 1

**Comment by iwasd** [comment]
hi
I face same issue
naabu use v2.0.4

naabu -host 192.168.0.101 -p 443,80 -stats -nmap-cli 'nmap -sV -oX naabu-output.xml'

[INF] Running nmap command: nmap -sV -oX naabu-output.xml -p 443,80 192.168.0.101
Starting Nmap 7.91 ( https://nmap.org ) at 2021-07-29 12:19 UTC
[ERR] Could not get network interfaces: exit status 1

**Comment by Mzack9999** [comment]
@dogasantos, I think that the port parameter is empty because naabu couldn't find any open port. This is yet another problem, as nmap shouldn't be run on an empty result, we will fix it.

@Shivangx01b @iwasd I'm trying to reproduce the issue, but it seems like naabu is always correctly running nmap on my box. Both suggested nmap commands seems syntactically correct, but their execution ends with error code 1:

```
nmap -sVC -Pn -oX /tmp/output.xml -p 80,443,8443,8080 104.16.100.52
nmap -sV -oX naabu-output.xml -p 443,80 192.168.0.101
```

I'm wondering if the nmap command run directly from the shell executes correctly?

**Comment by dogasantos** [comment]
@Mzack9999 yes, it might be the case, i'll test that ! Thanks for the update!

**Comment by dogasantos** [comment]
it seems to be the case

**Comment by ehsandeep** [comment]
This is now fixed in the dev version with linked PR https://github.com/projectdiscovery/naabu/pull/208
